runtimes/runtimes/util/serverDataFolder.test.ts

```diff
@@ -0,0 +1,190 @@
+import { getServerDataFolder } from './serverDataFolder'
+import assert from 'assert'
+import sinon from 'sinon'
+import { InitializeParams } from '../../protocol'
+import path from 'path'
+import os from 'os'
+
+describe('serverDataFolder', () => {
+    let initializeParams: InitializeParams
+    let pathJoinStub: sinon.SinonStub
+    let platformStub: sinon.SinonStub
+    let osHomeDirStub: sinon.SinonStub
+    const serverName = 'testServer'
+    const expectedClientFolderName = 'aws_vscode_sample_extension_for_vscode'
+
+    beforeEach(() => {
+        initializeParams = {
+            processId: null,
+            rootUri: null,
+            clientInfo: {
+                name: '.params VSCode',
+                version: '2.5.2',
+            },
+            capabilities: {},
+            initializationOptions: {
+                aws: {
+                    clientInfo: {
+                        name: 'aws VSCode ',
+                        version: '2.5.2',
+                        extension: {
+                            name: 'Sample Extension for VSCode',
+                            version: '0.0.1',
+                        },
+                    },
+                },
+            },
+        }
+        osHomeDirStub = sinon.stub(os, 'homedir').returns('stubHomeDir')
+    })
+
+    afterEach(() => {
+        sinon.restore()
+        process.env.APPDATA = ''
+        process.env.XDG_DATA_HOME = ''
```

Comment:
nit: we could improve this by using something like `mock-env` for mock env vars per test